api/pkg/transformer/symbol/time.go

```diff
@@ -0,0 +1,241 @@
+			tsInt64, err := converter.ToInt64(val.Interface())
+			if err != nil {
+				panic(err)
+			}
+
+			if timezoneVals.Kind() == reflect.Slice {
+				timeLocation, err := LoadLocationCached(fmt.Sprint(timezoneVals.Index(idx)))
+				if err != nil {
+					panic(err)
+				}
+				location = timeLocation
+			}
+
+			values = append(values, timeTransformerFn(tsInt64, location))
+		}
+		return values
+	default:
+		tsInt64, err := converter.ToInt64(ts)
+		if err != nil {
+			panic(err)
+		}
+
+		timeLocation, err := LoadLocationCached(fmt.Sprintf("%s", tz))
+		if err != nil {
+			panic(err)
+		}
+
+		return timeTransformerFn(tsInt64, timeLocation)
+	}
+}
+
+// ParseDateTime converts datetime given with specified format layout (e.g. RFC3339) into time.
+// It uses Golang provided time format layout (https://pkg.go.dev/time#pkg-constants).
+//
+// Examples:
+// Request JSON: {"booking_time": "2021-12-01 11:30:00", "timezone": "Asia/Jakarta"}
+// Expression: ParseDateTime($.booking_time, $.timezone, "2006-01-02 15:04:05")
+// Output: 2021-12-01 11:30:00 +07:00
+func (sr Registry) ParseDateTime(datetime, timezone interface{}, format string) interface{} {
+	if format == "" {
+		panic("datetime format must be specified")
+	}
+
+	dt, err := sr.evalArg(datetime)
+	if err != nil {
+		panic(err)
+	}
+
+	tz, err := sr.evalArg(timezone)
+	if err != nil {
+		panic(err)
+	}
+
+	dateTimeVals := reflect.ValueOf(dt)
+	timezoneVals := reflect.ValueOf(tz)
+
+	switch dateTimeVals.Kind() {
+	case reflect.Slice:
+		if dateTimeVals.Len() == 0 {
+			panic("empty array of timestamp provided")
+		}
+
+		location := &time.Location{}
+		if timezoneVals.Kind() == reflect.Slice {
+			if timezoneVals.Len() == 0 {
+				panic("empty array of timezone provided")
+			}
+
+			if dateTimeVals.Len() != timezoneVals.Len() {
+				panic("both date time and timezone arrays must have the same length")
+			}
+		} else {
+			timeLocation, err := LoadLocationCached(fmt.Sprintf("%s", tz))
+			if err != nil {
+				panic(err)
+			}
+			location = timeLocation
+		}
+
+		var values []interface{}
+		for idx := 0; idx < dateTimeVals.Len(); idx++ {
+			dt := dateTimeVals.Index(idx)
+
+			if timezoneVals.Kind() == reflect.Slice {
+				timeLocation, err := LoadLocationCached(fmt.Sprint(timezoneVals.Index(idx)))
+				if err != nil {
+					panic(err)
+				}
+				location = timeLocation
+			}
+
+			dateTime, err := time.ParseInLocation(fmt.Sprintf("%v", format), fmt.Sprintf("%v", dt), location)
+			if err != nil {
+				panic(err)
+			}
+
+			values = append(values, dateTime)
+		}
+		return values
+	default:
+		timeLocation, err := LoadLocationCached(fmt.Sprintf("%v", timezone))
+		if err != nil {
+			panic(err)
+		}
+
+		dateTime, err := time.ParseInLocation(fmt.Sprintf("%v", format), fmt.Sprintf("%v", dt), timeLocation)
+		if err != nil {
+			panic(err)
+		}
+
+		return dateTime
+	}
+}
+
+type LocationCache map[string]*time.Location
+
+var (
+	locationCache     = LocationCache{}
```

Comment:
Shall we use sync.Map instead? Based on the benchmark (write less, read more) the performance is the best. https://www.fatalerrors.org/a/map-concurrency-performance-and-principle-analysis-of-golang.html